wave_optimization/include/wave/optimization/factor_graph/variable.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef WAVE_OPTIMIZATION_FACTOR_GRAPH_VARIABLE_HPP
+#define WAVE_OPTIMIZATION_FACTOR_GRAPH_VARIABLE_HPP
+
+#include "wave/utils/utils.hpp"
+
+namespace wave {
+
+struct Variable {
+    size_t id;
```

Comment:
Only use `size_t` for sizes.. I actually thought all the `size_t` arguments in this PR were sizes, when they referred to "id"